src/test/scala/SystemTests.scala

```diff
@@ -162,16 +165,18 @@ trait SystemTests extends AnyFunSuite {
 }
 
 class SystemTestsBAP extends SystemTests  {
-  runTests(correctPrograms, correctPath, "correct", true, true)
-  runTests(incorrectPrograms, incorrectPath, "incorrect", false, true)
+  runTests(correctPrograms, correctPath, "correct", true, true, false)
+  runTests(incorrectPrograms, incorrectPath, "incorrect", false, true, false)
+  runTests(correctWithAnalysisPrograms, correctWithAnalysisPath, "correct", true, true, true)
   test("summary") {
     summary("testresult-BAP.csv")
   }
 }
 
 class SystemTestsGTIRB extends SystemTests  {
-  runTests(correctPrograms, correctPath, "correct", true, false)
-  runTests(incorrectPrograms, incorrectPath, "incorrect", false, false)
+  runTests(correctPrograms, correctPath, "correct", true, false, false)
+  runTests(incorrectPrograms, incorrectPath, "incorrect", false, false, false)
+  runTests(correctWithAnalysisPrograms, correctWithAnalysisPath, "correct", true, false, true)
```

Comment:
This (and the BAP one) should use "correct-analysis" since that's the name of the folder the tests are in.